pkgs/servers/rundeck/default.nix

```
@@ -0,0 +1,74 @@
+{
+  lib,
+  stdenv,
+  fetchurl,
+  makeWrapper,
+  jdk17,
+  which,
+  coreutils,
+  openssh,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "rundeck";
+  version = "5.7.0-20241021";
+
+  src = fetchurl {
+    url = "https://packagecloud.io/pagerduty/rundeck/packages/java/org.rundeck/rundeck-${version}.war/artifacts/rundeck-${version}.war/download?distro_version_id=167";
+    hash = "sha256-wKpdAk3sd7mvuD5LKuBT2M+Y2B5KOKmhiiwE1krpKyw=";
+  };
+
+  nativeBuildInputs = [ makeWrapper ];
+  buildInputs = [ jdk17 ];
+
+  dontUnpack = true;
+
+  installPhase = ''
+    mkdir -p $out/share/rundeck
+    cp $src $out/share/rundeck/rundeck.war
+
+    mkdir -p $out/bin
+    # Main rundeck executable
+    makeWrapper ${jdk17}/bin/java $out/bin/rundeck \
+      --set RDECK_BASE "/var/lib/rundeck" \
+      --add-flags "-Xmx4g" \
+      --add-flags "-Drdeck.base=/var/lib/rundeck" \
+      --add-flags "-Drundeck.config.location=/etc/rundeck" \
+      --add-flags "-jar $out/share/rundeck/rundeck.war" \
+      --prefix PATH : ${
+        lib.makeBinPath [
+          which
+          coreutils
+          openssh
+        ]
+      }
+
+    # Installation helper script
+    makeWrapper ${jdk17}/bin/java $out/bin/rundeck-install \
+      --set RDECK_BASE "/var/lib/rundeck" \
+      --add-flags "-jar $out/share/rundeck/rundeck.war" \
+      --add-flags "--installonly" \
+      --prefix PATH : ${
+        lib.makeBinPath [
+          which
+          coreutils
+          openssh
+        ]
+      }
+  '';
+
+  meta = with lib; {
+    description = "Job scheduler and runbook automation";
+    longDescription = ''
+      Rundeck is an open source automation service with a web console,
+      command line tools and a WebAPI. It lets you easily run automation tasks
+      across a set of nodes.
+    '';
+    homepage = "https://www.rundeck.com/";
+    changelog = "https://docs.rundeck.com/docs/history/";
+    sourceProvenance = with sourceTypes; [ binaryBytecode ];
+    license = licenses.asl20;
+    platforms = platforms.unix;
+    maintainers = with maintainers; [ liberodark ];
+  };
```

```suggestion
  meta = {
    description = "Job scheduler and runbook automation";
    longDescription = ''
      Rundeck is an open source automation service with a web console,
      command line tools and a WebAPI. It lets you easily run automation tasks
      across a set of nodes.
    '';
    homepage = "https://www.rundeck.com/";
    changelog = "https://docs.rundeck.com/docs/history/";
    sourceProvenance = with lib.sourceTypes; [ binaryBytecode ];
    license = lib.licenses.asl20;
    platforms = lib.platforms.unix;
    maintainers = with lib.maintainers; [ liberodark ];
  };
```
Avoid using the `with` pattern.